hcpannot/analysis/contour.py

```
@@ -0,0 +1,253 @@
+    # for each subject, try to load the contour coordinates
+    for sid in sids:
+        cache_file = os.path.join(proc_path, space, f'cacherater_{rater}_{sid}_{hemi}_{roi}.mgz')
+        
+        # check if the contour coordinates are already saved
+        if os.path.isfile(cache_file):
+            
+            # if coordinates are saved, load them
+            coords = ny.load(cache_file)
+            
+            # x and y coordinates were saved as separate arrays (npoints * 2)
+            # stack them together to create a list of coordinates
+            lines.append(np.column_stack(coords))
+        
+        else:
+            # if the coordinates are not saved, add the subject to the missing list
+            missing_contour.append(sid)
+            
+    lc = LineCollection(lines, linewidths=lw, alpha=alpha, colors=color)
+    return (lc, missing_contour)
+
+def plot_lc(sids, hemi, rois, space, proc_path, rater=meanrater,
+            plot_v123=False, meanlines=None, npoints=500,flatmap=True, 
+            ax=None, lw=0.25, alpha=0.3, colors=None):
+    """
+    Plot a LineCollection object on a flatmap. By default, the contours are plotted on
+    flatmap with the mean V1-V3 contours.
+    
+    """
+    if ax is None:
+        fig, ax = plt.subplots(1,1, figsize=(3.5,3.5), dpi=1200)
+    
+    if not isinstance(rois, list):
+        rois = [rois]
+        
+    if plot_v123:
+        if meanlines is None:
+            print('Mean V1-V3 contours not provided.')
+            return
+        else:
+            meanv123 = meanlines[hemi]
+            for t in meanv123.keys():
+                for c in meanv123[t]:
+                    trace = meanv123[t][c].curve.linspace(npoints)
+                    
+                    if hemi == 'lh':
+                        ax.plot(trace[0]+8, trace[1], 'k-') # shift the x coordinates by 8
+                    else:
+                        ax.plot(trace[0]-8, trace[1], 'k-') # shift the x coordinates by -8
```

Might want to raise an explicit error when `meanlines` isn't a dict.